Source/Core/Core/HW/SI/SI.cpp

```diff
@@ -365,27 +351,29 @@ void Init()
     s_channel[i].out.hex = 0;
     s_channel[i].in_hi.hex = 0;
     s_channel[i].in_lo.hex = 0;
+    s_channel[i].has_recent_device_change = false;
 
     if (Movie::IsMovieActive())
     {
+      s_desired_device_types[i] = SIDEVICE_NONE;
+
       if (Movie::IsUsingPad(i))
       {
         SIDevices current = SConfig::GetInstance().m_SIDevice[i];
         // GC pad-compatible devices can be used for both playing and recording
         if (SIDevice_IsGCController(current))
```

Comment:
Kinda wondering if we could/should skip the `SIDevice_IsGCController` call if `Movie::IsUsingBongo(i)` is true already and do something like this:
```c++
if (Movie::IsUsingBongo(i))
  s_desired_device_types[i] = SIDEVICE_GC_TARUKONGA;
else if (SIDevice_IsGCController(current))
  s_desired_device_types[i] = current;
else
  s_desired_device_types[i] = SIDEVICE_GC_CONTROLLER;
```
(...and now that it is written like that; does this actually make sense, defaulting to a GC Controller?)

Or if we don't want that, the `else` (and the `if`) should probably have curlies since it breaks across multiple lines now...

Reply:
Yeah. The TAS logic really makes no sense to me. I tried to leave the behavior unchanged.